apps/api/views.py

```diff
@@ -373,7 +373,86 @@ def consent(request: Request):
     participant_data = get_object_or_404(
         ParticipantData, system_metadata__commcare_connect_channel_id=request_data["channel_id"]
     )
-    participant_data.system_metadata["consent"] = request_data["consent"]
-    participant_data.save(update_fields=["system_metadata"])
+    participant_data.update_consent(request_data["consent"])
 
     return HttpResponse()
+
+
+@extend_schema(
+    operation_id="trigger_bot_message",
+    summary="Trigger the bot to send a message to the user",
+    tags=["Channels"],
+    request=TriggerBotMessageRequest(),
+    responses={
+        200: {},
+        400: {"description": "Bad Request"},
+        404: {"description": "Not Found"},
+    },
+    examples=[
+        OpenApiExample(
+            name="GenerateBotMessageAndSend",
+            summary="Generates a bot message and sends it to the user",
+            value={
+                "identifier": "part1",
+                "experiment": "exp1",
+                "platform": "connect_messaging",
+                "prompt_text": "Tell the user to do something",
+            },
+            status_codes=[200],
+        ),
+        OpenApiExample(
+            name="ParticipantNotFound",
+            summary="Participant not found",
+            value={"detail": "Participant not found"},
+            status_codes=[404],
+        ),
+        OpenApiExample(
+            name="ExperimentChannelNotFound",
+            summary="Experiment cannot send messages on the specified channel",
+            value={"detail": "Experiment cannot send messages on the connect_messaging channel"},
+            status_codes=[404],
+        ),
+        OpenApiExample(
+            name="ConsentNotGiven",
+            summary="User has not given consent",
+            value={"detail": "User has not given consent"},
+            status_codes=[400],
+        ),
+    ],
+)
+@api_view(["POST"])
+def trigger_bot_message(request):
+    """
+    Trigger the bot to send a message to the user
+    """
+    serializer = TriggerBotMessageRequest(data=request.data)
+    serializer.is_valid(raise_exception=True)
+
+    identifier = serializer.data["identifier"]
+    platform = serializer.data["platform"]
+    experiment_public_id = serializer.data["experiment"]
+
+    experiment = get_object_or_404(Experiment, public_id=experiment_public_id, team=request.team)
+
+    try:
+        participant_data = ParticipantData.objects.defer("data").get(
+            participant__identifier=identifier,
+            participant__platform=platform,
+            object_id=experiment.id,
+            content_type=ContentType.objects.get_for_model(Experiment),
+        )
+
+        ExperimentChannel.objects.filter(platform=platform, experiment=experiment).exists()
```

Comment:
this won't raise an exception now so you'll need to check that the return value is `True`

Reply:
https://github.com/dimagi/open-chat-studio/pull/1041/commits/0b87ceb0a93bedd5d97331a9f88a2ee4555b8a1a